### PR TITLE
AssignmentExpr is not a BindableExpression

### PR DESCRIPTION
**Pull request by hauntsaninja**

### CODE REVIEW

File: test-data/unit/check-python38.test [hunk 1]
```diff
@@ -403,7 +403,14 @@ if (thing := get_thing()).is_good:
     reveal_type(thing)  # N: Revealed type is "__main__.Good"
 else:
     reveal_type(thing)  # N: Revealed type is "__main__.Bad"
-[builtins fixtures/property.pyi]
+
+def get_things() -> Union[Tuple[Good], Tuple[Bad]]: ...
+
+if (things := get_things())[0].is_good:
```
Comment (sobolevn) [inline test-data/unit/check-python38.test:409]: You have modified two branches for `IndexExpr` and `MemberExpr`, but this new test covers `NameExpr`. Do we have other two covered?

Comment (hauntsaninja) [reply test-data/unit/check-python38.test:409]: Good point, I'll fix this tomorrow. Although actually maybe we should do the walrus collapse inside the binder put

Comment (sobolevn) [reply test-data/unit/check-python38.test:409]: The tomorrow has just started in my timezone 😆

Comment (hauntsaninja) [reply test-data/unit/check-python38.test:409]: > this new test covers NameExpr

The new test actually covers IndexExpr in the branch above.

I tried adding tests with IndexExpr and MemberExpr inside the walrus, but it turns out that's not valid Python

**Review by sobolevn — APPROVED** [review]

**Pull request merged**